### DISCUSSION TITLE
OpenPlotter Maiana App refuses to see the device

### DISCUSSION THREAD

**Discussion by palunanue** [peterantypas/maiana#111] [Q&A]
This is a head-scratcher. I've installed everything according to instructions, and individually, everything seems to work. The Maiana AIS has green power, green GPS and RX flickers every now and then. The NMEA2000 connection delivers GPS and AIS targets which OpenCPN can see just fine. With MiniCom on the Raspberry PI with OpenPloter (I don't have a PC for terminal usage) I can see the RX traffic just fine. I cannot for the life of me find out how to properly send the CLI command in MiniCom, seems like Termite on PC is the only app that can do that properly. So I try to provision the Maiana through the OpenPlotter App. 

For the longest time it simply didn't show in the list of apps, even after updating and refreshing all the time. Today, however, it showed up so I tried to provision. I followed all the steps integrating from Serial into Signal K. USB - USB with Serial 38.4k 8N1. The Maiana shows up just fine in Signal K, and the data browser show Maiana-specific data such as AIS strength and Maiana FW version. Signal K has the connection to the Maiana all the time, unless I'm testing with MiniCom but I make sure to release the port. The Maiana Transponder App has been authenticated with Signal K, and if I change the R/W permissions from R/W to e.g. R, the app complains that it needs R/W.

So. The Signal K is talking (or at least listening) to the Maiana. The Maiana is working perfectly with NMEA2000 network. The Maiana Transponder app 2.1.1 seems to talk fine with Signal K.

But, no matter what I do in the app it says the device is not connected, try to refresh. I've tried refresing a million times, reinstalling the Transponder app, remiving and reinstalling the Signal K connection, removing and reinstalling the app integration to signal K, nothing will make the app see the device. All the time, the device is happily transmitting RX signals over both the NMEA2000 network, and into the terminal using USB connection.

What is going on? What am I missing?

**Comment by sailoog** [comment]
OpenPlotter version?
openplotter-maiana app version?

enable "debugging" in openplotter-settings and type openplotter-maiana in a terminal to open the app. Try again and copy here any error.

**Reply by palunanue** [reply]
Good suggestions, will collect that info when I'm back at the boat in a few days. Your question response time is awesome as always. :)

**Reply by paaluz** [reply]
openplotter-maiana version 2.1.2
OpenCPN version 5.6.2
Settings version 2.9.1
Errors in terminal when running openplotter-maiana:

> pi@openplotter:~ $ openplotter-maiana
> 
> (openplotter-maiana:2680): Gtk-CRITICAL **: 19:11:48.213: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar
> 
> (openplotter-maiana:2680): Gtk-CRITICAL **: 19:11:48.456: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar
> invalid literal for int() with base 10: 'x'
> invalid literal for int() with base 10: 'x'
> invalid literal for int() with base 10: 'x'
> invalid literal for int() with base 10: 'x'

**Reply by paaluz** [reply]
The "invalid literal for int() with base 10: 'x'" error comes every time the refresh button is clicked.

**Reply by sailoog** [reply]
You can ignore Gtk errors but that invalid literal seems to be the problem, could you send me your transmission settings by private to info[you-know-what]openmarine.net?

edited: wait, if I am not wrong you could not set the transmission data yet. The issue could be when your firmware version is parsed, could you post here the hardware version, firmware version, type of MCU and serial number?

**Reply by palunanue** [reply]
Let's continue this threat over at OpenMarine.org: [https://forum.openmarine.net/showthread.php?tid=4117](https://forum.openmarine.net/showthread.php?tid=4117)

**Comment by sailoog** [comment]
@peterantypas 

He is getting "11.x" as hardware revision and OpenPlotter is expecting a number instead off that x. Is this intentional?

**Comment by peterantypas** [comment]
This is probably an old board. It’s not compatible.

**Reply by paaluz** [reply]
So what do I do now? It was the replacement board that replaced one not able to enter into CLI mode at all. Also, when enabling TX mode, there is no data sent as far as I can see. Nothting shows up in MarineTraffic, even after waiting a long time in good coverage.

**Reply by peterantypas** [reply]
Can you confirm the actual board revision printed on the PCB? You may have bad firmware as well.

**Reply by paaluz** [reply]
Board rev is 6.0

Image uploaded elsewhere in this thread.

**Comment by peterantypas** [comment]
Where did you obtain this board? It's not compatible with OpenPlotter, did
I ship it to you?

**Reply by paaluz** [reply]
Yes, I'm the one that emailed you about a Maiana that flaked out. It would not enter CLI mode and the device lights started flickering in a weird way. You sent a replacement board and that's the one I'm referring to here. :)

**Reply by paaluz** [reply]
![AEB8FC5A-B19A-4976-A773-37A8BA06759C](https://user-images.githubusercontent.com/107363102/176250562-ef7ef5e5-f318-4931-b68a-697e8b2f6d5f.jpeg)

**Reply by peterantypas** [reply]
I was asking about the main transponder board revision, not the adapter. Put it in CLI mode and paste the response to the sys? command, I want to see the entire string.

**Comment by paaluz** [comment]
![image](https://user-images.githubusercontent.com/107363102/176259034-5f413227-64f9-491e-bbaa-417eaeec1e65.jpg)

**Reply by peterantypas** [reply]
OK. This appears to be board rev 11.3.0. Unfortunately, some of those boards were built with 2 different MCUs: L422 and L432. And this one has really old firmware and it's not reporting that. Is there a way to physically inspect it or is it mounted? Depending on which MCU you have, I should be able to make a new firmware image that will bring it into compliance with OpenPlotter.

**Reply by paaluz** [reply]
It’s pretty firmly mounted but I did take a few snaps of it upon arrival. See if yo can make out the info. 
![088619AC-585D-4531-AD3E-889459B3FC3D](https://user-images.githubusercontent.com/107363102/176288398-146d62c6-5360-4e20-83f2-d7e64af9271f.jpeg)
![7353AAC8-3402-4B5F-85F4-1DAE00341595](https://user-images.githubusercontent.com/107363102/176288401-fb09a76e-0021-4459-8f61-49fd1807af91.jpeg)

**Comment by peterantypas** [comment]
The board is 11.3.0, most likely with an L422 MCU. Firmware version 3.3.1
is old and buggy (it doesn't transmit).

First thing is to update its firmware. Follow the instructions here and use
the attached binary.

https://github.com/peterantypas/maiana/blob/master/latest/Manuals/MAIANA%E2%84%A2%20Firmware%20Update.pdf

After board updates and reboots, put it in CLI mode again and send this
command:

otp 11.3.0

This should bring this board into compliance with OpenPlotter.

**Reply by paaluz** [reply]
Thanks! But I cannot se the compiled binary. Where is it attached?

**Reply by peterantypas** [reply]
They don't like .bin attachments here. This zip should work.

[maiana-stm32l422-11.3.0-v4.1.0-fullimage.zip](https://github.com/peterantypas/maiana/files/9011003/maiana-stm32l422-11.3.0-v4.1.0-fullimage.zip)

**Reply by peterantypas** [reply]
You have to unzip it of course

**Reply by paaluz** [reply]
I’ll fire up my unlicensed version of PKZip and report back. ;)

**Reply by paaluz** [reply]
Ok, everything went well with the update. Maiana now talks just fine with OpenPlotter and I can see the data. But even though it says it’s transmitting, it does now show up in MarineTraffic. 

Attached photos for reference. 

![image](https://user-images.githubusercontent.com/107363102/176508742-d437d143-b612-4286-9c17-b5362383b7c8.jpg)

![image](https://user-images.githubusercontent.com/107363102/176508824-2fc05f11-9c19-49d0-bc87-627cfc94e258.jpg)
![D1D3B968-829C-4BF0-BA70-F3519A322DB5](https://user-images.githubusercontent.com/107363102/176508885-49aeb14e-a78a-4fdc-99ce-9119de7f80a9.jpeg)

**Reply by kagouraki** [reply]
#107 check here. Had the same problem

**Reply by paaluz** [reply]
@kagouraki Thanks for the tip! Unfortunlately the TX light is quite static on and does not blink nor flicker. I've captured some traffic on MiniCom and there are as far as I can search in the traffic, no $PAITX sentences.

**Reply by kagouraki** [reply]
Maybe you have an invalid string. What are you sending to the cli command to programm? Paste the string here

**Reply by paaluz** [reply]
String to provision the unit is: "station nnnnnnnnn,NNNNNNNNNNNN,SSssss,36,12,4,3,11" 
Where n is MMSI, N is 12 captial characters and one number at the end without space or 
punctuation, SS is two letters and ssss is four numbers.

I'm seeing some PAITX strings now:
$PAITX,B,18*1F
$PAITX,A,18*1C
$PAITX,A,18*1C
$PAITXCFG,1,0,1,1,0*0B

**Reply by paaluz** [reply]
The action taken is to send the 'station' string again after firmware update, and now the PAITX strings appear. Nothing in Marinetraffic though.  No change in the TX light it seems firmly on. How often is it supposed to flicker?

**Reply by paaluz** [reply]
Hm. The output is slightly "off" now. 

[00][00][00][00][00][06][0C]vÿÿÿÿÿ[0C][17][00][00][04]v[04]$GNGLL,5827.7047,N,01112.8835,E,113036.000,A,D*4D
01112.8852,E,4.97,34.00,300622,,,D,V*37
$GPVTG,34.00,T,,M,4.97,N,9.21,K,D*0F
3,36,75,26,251,32,70,13,161,21,7,M,,*40
3037.000,5827.7059,N,01112.8852,E,2,14,0.65,-4.3,M,40.7N,9.21,K,$GNGSA,A,3,10,23,16,32,15,18,27,07,21,30,,,1.14,0.65,0.93,1*05
$GNGSA,A,3,85,75,68,76,,,,,,,,,1.14,0.65,0.93,2*0F
$GPGSV,4,1,14,27,68,162,33,10,62,118,32,23,45,065,37,21,33,258,31,0*6E
$GPGSV,4,2,14,36,21,156,36,16,17,191,22,15,15,031,40,30,10,314,22,0*6D
!AIVDM,1,1,,A,B3uIeB00@p<oHl`FIp@7kwjQl5Ip,0*24
8,08,078,39,07,06,287,26,32,05,136,39,14,04,336,29,0*6D
,3,14,12,0*6D
$GPGSV,4,4,14,13,04,001,28,01,03,256,30,0*6D
!AIVDM,1,1,,B,B39roNP0;0<q?j`Fp0Is7wjUkP06,0*53
$GLGSV,2,1,08,85,57,294,32,84,53,083,22,69,48,101,34,76,33,316,32,1*73
,251,31,70,13,161,21,77,08,000,,1*73
2,69,48,101,34,76,33,316,32,1*73$GNGLL,5827.7059,N,01112.8852,E,113037.000,A,D*42
$GNRMC,113038.000,A,5827.7071,N,01112.8866,E,4.97,34.40,300622,,,D,V*31
$GPVTG,34.40,T,,M,4.97,N,9.22,K,D*08
$GPGGA,113038.000,5827.7071,N,01112.8866,E,2,14,0.65,-4.3,M,40.7,M,,*42
$GNGSA,A,3,10,23,16,32,15,18,27,07,21,30,,,1.13,0.65,0.93,1*02
!AIVDM,1,1,,B,B39QOpP0>H<jek8F=G3M;wk1lD18,0*00
$GNGSA,A,3,85,75,68,76,,,,,,,,,1.13,0.65,0.93,2*08
$GPGSV,4,1,14,27,68,162,34,10,62,118,42,23,45,065,38,21,33,258,30,0*60
,191,24,15,15,031,41,30,10,314,23,0*6A
23,45,065,38,21,33,258,30,0*60
$GPGSV,4,3,14,18,08,078,38,07,06,287,26,32,05,136,39,14,04,336,29,0*6C
$GPGSV,4,4,14,13,04,001,27,01,03,256,30,0*62
$GLGSV,2,1,08,85,57,294,33,84,53,083,22,69,48,101,34,76,33,316,33,1*73
$GLGSV,2,2,08,68,31,033,35,75,26,251,29,70,13,161,21,77,08,000,,1*79
$GNGLL,5827.7071,N,01112.8866,E,113038.000,A,D*40
!AIVDM,1,1,,B,13ufUR70000kL1pQIMQMTnap0pFw,0*71
01112.8883,E,5.03,35.48,300622,,,D,V*33
w,0*71
$GNRMC,113039.000,A,5827$GPVTG,35.48,T,,M,5.03,N,9.32,K,D*0C
112.8883,E,2,14,0.65,-4.2,M,40.7,M,,*44
$GNGSA,A,3,10,23,16,32,15,18,27,07,21,30,,,1.13,0.65,0.93,1*02
[00][00][00][00][00][00][00][00][00][00][00][00][00][00][00][00][00][00][00]˜iX[02][00][00][00][00]ÿÿÿÿ[0C]$GNGSA,A,3,85,75,68,76,,,,,,,,,1.13,0.65,0.93,2*08
$GPGSV,4,1,14,27,68,162,34,10,62,118,41,23,45,065,27,21,33,258,30,0*6D
$GPGSV,4,2,14,36,21,156,38,16,17,191,23,15,15,031,41,30,10,314,23,0*62
,287,26,32,05,136,39,14,04,336,29,0*6C
$GPGSV,4,4,14,13,04,001,27,01,03,256,30,0*62
13,0.65,0.93,1*02
[00][00][00][00][00][00][00][00][00][00][00][00][00]$GLGSV,2,1,08,85,57,294,34,84,53,083,22,69,48,101,35,76,33,316,33,1*75
$GLGSV,2,2,08,68,31,033,33,75,26,251,29,70,13,161,21,77,08,000,,1*7F
$GNGLL,5827.7083,N,01112.8883,E,113039.000,A,D*47
$GNRMC,113040.000,A,5827.7095,N,01112.8899,E,5.02,39.79,300622,,,D,V*3E
$GPVTG,39.79,T,,M,5.02,N,9.30,K,D*01

**Reply by muebau** [reply]
> The action taken is to send the 'station' string again after firmware update, and now the PAITX strings appear. Nothing in Marinetraffic though. No change in the TX light it seems firmly on. How often is it supposed to flicker?

The station in Heestrand should have received your signal easily. Do other ships nearby see your signal?

**Reply by kagouraki** [reply]
You need to have another reciever to see your transmission. Marine trafic does not show you. I tried in three transfers and nothing. But the other ships around recieved my ais. 
Transmission every 5 minutes when stationary.
Every 2 minutes i think when moving in sailboat speeds.

**Reply by palunanue** [reply]
Hey guys! I just wanted to thank you all for good help in solving these problems. You were right, the device was indeed TXing it just tool MarineTraffic about a week to become aware of it... :) The device now turns on whern the NMEA2000 network is running, so every time we use the for for a bit. Love this project!
